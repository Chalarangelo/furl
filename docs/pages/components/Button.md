---
title: Button
description: furl's Button component.
---

Buttons allow users to interact with your application.

### Examples

<buttonexamples></buttonexamples>

```jsx
import React from 'react';
import ReactDOM from 'react-dom';
import { Button } from 'furl-components';

ReactDOM.render(
  <div>
    <Button 
      color='primary' 
      size='large'
      fill='solid'
      text='uppercase'
      onClick={() => console.log('Hello!')}
    >
      Hello
    </Button>
    <Button
      type='link'
      fill='link'
      shape='rounded'
      href='https://www.google.com/'
      openIn='external'
    >
      Goodbye
    </Button>
  </div>, 
  document.getElementById('root')
);
```

### API

<table>
  <caption>Button props</caption>
  <thead>
    <tr>
      <th>Property</th>
      <th colspan="3">Description</th>
      <th>Type</th>
      <th>Default</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td class="font-c">color</td>
      <td colspan="3">button color</td>
      <td>string</td>
      <td class='font-c'>'plain'</td>
    </tr>
    <tr>
      <td class="font-c">size</td>
      <td colspan="3">button size</td>
      <td>string</td>
      <td class='font-c'>'normal'</td>
    </tr>
    <tr>
      <td class="font-c">fill</td>
      <td colspan="3">button fill style</td>
      <td>string</td>
      <td class='font-c'>'flat'</td>
    </tr>
    <tr>
      <td class="font-c">shape</td>
      <td colspan="3">button shape</td>
      <td>string</td>
      <td class='font-c'>'normal'</td>
    </tr>
    <tr>
      <td class="font-c">text</td>
      <td colspan="3">button text transformation</td>
      <td>string</td>
      <td class='font-c'>'normal'</td>
    </tr>
    <tr>
      <td class="font-c">type</td>
      <td colspan="3">button type</td>
      <td>string</td>
      <td class='font-c'>'button'</td>
    </tr>
    <tr>
      <td class="font-c">disabled</td>
      <td colspan="3">button disabled state</td>
      <td>boolean</td>
      <td class='font-c'>false</td>
    </tr>
    <tr>
      <td class="font-c">href</td>
      <td colspan="3">button hyperlink url</td>
      <td>string</td>
      <td class='font-c'></td>
    </tr>
    <tr>
      <td class="font-c">openIn</td>
      <td colspan="3">target of the hyperlink></td>
      <td>string</td>
      <td class='font-c'>'self'</td>
    </tr>
    <tr>
      <td class="font-c">onClick</td>
      <td colspan="3">button onclick event></td>
      <td>function</td>
      <td class='font-c'></td>
    </tr>
  </tbody>
</table>


### Notes

* `color` can be any of the predefined color palettes (`'plain'`, `'primary'`, `'secondary'`, `'success'`, `'warning'` or `'danger'`).
* `size` can be one of the following: `'small'`, `'normal'` or `'large'`.
* `fill` can be one of the following: `'flat'`, `'solid'`, `'light'`, `'gradient'`, `'outline'`, `'ghost'` or `'link'`.
* `shape` can be either `'normal'` or `'rounded'`.
* `text` can be either `'normal'` or `'uppercase'`.
* `type` can be one of the following: `'button'`, `'submit'`, `'reset'` or `'link'`.
* `href` and `openIn` only apply when `type` is equal to `'link'`.
