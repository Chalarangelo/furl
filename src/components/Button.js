import React from "react";

const Button = props => <button>{props.label}</button>;

export default Button;