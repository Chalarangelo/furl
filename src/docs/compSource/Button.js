export default `import { Button } from furl-components;

ReactDOM.render(<div>
  <Button color="plain">Default</Button>
  <Button color="primary">Primary</Button>
  <Button color="secondary">Secondary</Button>
  <Button color="success">Success</Button>
  <Button color="warning">Warning</Button>
  <Button color="danger">Danger</Button>

  <Button fill="flat">Default</Button>
  <Button fill="solid">Solid</Button>
  <Button fill="light">Light</Button>
  <Button fill="gradient">Gradient</Button>
  <Button fill="outline">Outline</Button>
  <Button fill="ghost">Ghost</Button>
  <Button disabled>Disabled</Button>

  <Button size="small">Small</Button>
  <Button size="normal">Medium</Button>
  <Button size="large">Large</Button>

  <Button shape="normal">Normal</Button>
  <Button shape="rounded">Rounded</Button>
  <Button shape="circle">Circle</Button>
  <Button text="uppercase">Upper</Button>

  <Button type="link" href="https://google.com" openIn="external">Link</Button>

  <Button onClick={(e) => {console.log(e)}}>Event handling</Button>
</div>);
`;