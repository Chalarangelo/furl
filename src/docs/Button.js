import React from "react";
import Hyperlink from "../components/Hyperlink";
import { Table, TableCaption, TableHead, TableBody, TableRow, TableCell} from "../components/Table";

const ButtonDocs = props => (
  <div>
    <h2>Button</h2>
    <p>Buttons allow users to interact with your application.</p>
    <h3>Examples</h3>
    {/* TODO: Add component demo */}
    <h3>API</h3>
    <p>Buttons are highly customizable, providing a different visual style for every kind of interaction context.</p>
    <Table className="fixed-table">
      <TableCaption>Button props</TableCaption>
      <TableHead>
        <TableRow>
          <TableCell heading>Name</TableCell>
          <TableCell heading>Type</TableCell>
          <TableCell heading>Default</TableCell>
          <TableCell heading heading colSpan={3}>Description</TableCell>
        </TableRow>
      </TableHead>
      <TableBody>
        <TableRow>
          <TableCell><code>color</code></TableCell>
          <TableCell><code>string</code></TableCell>
          <TableCell><code>'plain'</code></TableCell>
          <TableCell colSpan={3}>color palette of the button (<code>'plain'</code>, <code>'primary'</code>, <code>'secondary'</code>, <code>'success'</code>, <code>'warning'</code>, <code>'danger'</code>)</TableCell>
        </TableRow>
        <TableRow>
          <TableCell><code>fill</code></TableCell>
          <TableCell><code>string</code></TableCell>
          <TableCell><code>'flat'</code></TableCell>
          <TableCell colSpan={3}>fill style of the button (<code>'flat'</code>, <code>'solid'</code>, <code>'light'</code>, <code>'gradient'</code>, <code>'outline'</code>, <code>'ghost'</code>)</TableCell>
        </TableRow>
        <TableRow>
          <TableCell><code>size</code></TableCell>
          <TableCell><code>string</code></TableCell>
          <TableCell><code>'normal'</code></TableCell>
          <TableCell colSpan={3}>size of the button (<code>'normal'</code>, <code>'small'</code>, <code>'large'</code>)</TableCell>
        </TableRow>
        <TableRow>
          <TableCell><code>shape</code></TableCell>
          <TableCell><code>string</code></TableCell>
          <TableCell><code>'normal'</code></TableCell>
          <TableCell colSpan={3}>shape of the button (<code>'normal'</code>, <code>'rounded'</code>, <code>'circle'</code>)</TableCell>
        </TableRow>
        <TableRow>
          <TableCell><code>text</code></TableCell>
          <TableCell><code>string</code></TableCell>
          <TableCell><code>'normal'</code></TableCell>
          <TableCell colSpan={3}>text style of the button (<code>'normal'</code>, <code>'uppercase'</code>)</TableCell>
        </TableRow>
        <TableRow>
          <TableCell><code>disabled</code></TableCell>
          <TableCell><code>boolean</code></TableCell>
          <TableCell><code>false</code></TableCell>
          <TableCell colSpan={3}>disabled state of button</TableCell>
        </TableRow>
        <TableRow>
          <TableCell><code>type</code></TableCell>
          <TableCell><code>string</code></TableCell>
          <TableCell><code>'button'</code></TableCell>
          <TableCell colSpan={3}>type of the button (<code>'button'</code>, <code>'link'</code>)</TableCell>
        </TableRow>
        <TableRow>
          <TableCell><code>href</code></TableCell>
          <TableCell><code>string</code></TableCell>
          <TableCell>-</TableCell>
          <TableCell colSpan={3}>hyperlink url (only applicable if <code>type='link'</code>)</TableCell>
        </TableRow>
        <TableRow>
          <TableCell><code>openIn</code></TableCell>
          <TableCell><code>string</code></TableCell>
          <TableCell>-</TableCell>
          <TableCell colSpan={3}>hyperlink target (<code>'self'</code>, <code>'blank'</code>, <code>'external'</code>, only applicable if <code>type='link'</code>)</TableCell>
        </TableRow>
        <TableRow>
          <TableCell><code>onClick</code></TableCell>
          <TableCell><code>function (event)</code></TableCell>
          <TableCell>-</TableCell>
          <TableCell colSpan={3}>callback function to handle the <code>click</code> event</TableCell>
        </TableRow>
        <TableRow>
          <TableCell><code>id</code></TableCell>
          <TableCell><code>string</code></TableCell>
          <TableCell>-</TableCell>
          <TableCell colSpan={3}>value for the rendered element's <code>id</code></TableCell>
        </TableRow>
        <TableRow>
          <TableCell><code>className</code></TableCell>
          <TableCell><code>string</code></TableCell>
          <TableCell>-</TableCell>
          <TableCell colSpan={3}>additional classes that should be passed to the component</TableCell>
        </TableRow>
        <TableRow>
          <TableCell><code>children</code></TableCell>
          <TableCell><code>React.children</code></TableCell>
          <TableCell>-</TableCell>
          <TableCell colSpan={3}>the component's children elements</TableCell>
        </TableRow>
      </TableBody>
    </Table>
  </div>
);

export default ButtonDocs;
