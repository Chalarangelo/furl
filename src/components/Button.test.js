import React from "react";
import Button from "./Button";
import renderer from "react-test-renderer";

describe("Button", () => {
  it("renders properly", () => {
    const tree = renderer
      .create(<Button label="Continue"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});