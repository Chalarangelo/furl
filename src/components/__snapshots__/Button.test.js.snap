// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders properly 1`] = `
<button>
  Continue
</button>
`;
